src/common/adapters/MultiAdapter.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {Auth} from "src/misc/Auth.sol";
+import {CastLib} from "src/misc/libraries/CastLib.sol";
+import {BytesLib} from "src/misc/libraries/BytesLib.sol";
+import {ArrayLib} from "src/misc/libraries/ArrayLib.sol";
+import {MathLib} from "src/misc/libraries/MathLib.sol";
+
+import {IMessageHandler} from "src/common/interfaces/IMessageHandler.sol";
+import {IAdapter} from "src/common/interfaces/IAdapter.sol";
+import {MessageProofLib} from "src/common/libraries/MessageProofLib.sol";
+import {IMultiAdapter, MAX_ADAPTER_COUNT} from "src/common/interfaces/IMultiAdapter.sol";
+
+/// @title  Axelar Adapter
+/// @notice Routing contract that integrates with an Axelar Gateway
+contract MultiAdapter is Auth, IMultiAdapter {
+    using CastLib for *;
+    using MessageProofLib for *;
+    using BytesLib for bytes;
+    using ArrayLib for uint16[8];
+    using MathLib for uint256;
+
+    uint8 public constant PRIMARY_ADAPTER_ID = 1;
+    uint256 public constant RECOVERY_CHALLENGE_PERIOD = 7 days;
+
+    uint16 public immutable localCentrifugeId;
+    IMessageHandler public immutable gateway;
+
+    mapping(uint16 centrifugeId => IAdapter[]) public adapters;
+    mapping(uint16 centrifugeId => mapping(IAdapter adapter => Adapter)) internal _activeAdapters;
+    mapping(uint16 centrifugeId => mapping(bytes32 batchHash => Inbound)) public inbound;
+    mapping(uint16 centrifugeId => mapping(IAdapter adapter => mapping(bytes32 painbound => uint256 timestamp))) public
```

Comment:
Nit: I don't understand the choice for using the word `painbound`. Could you explain? The previous `payloadHash` makes more sense to me.

Reply:
It was a typo! Will fix this

Reply:
Done